### DISCUSSION TITLE
Support of Jalali calendar

### DISCUSSION THREAD

**Discussion by zfarhad** [gpbl/react-day-picker#1825] [Ideas]
Hi

I want to use React DayPicker in my project but I need to support jalali calendar (a different type of solar calendar).
By default, DayPicker uses date-fns library to calculate and manipulate date objects in JS. There is a forked version of date-fns library for jalali calendar system. ([date-fns-jalali](https://www.npmjs.com/package/date-fns-jalali)) I have installed date-fns-jalali library and forced DayPicker components to use this one instead of the date-fns library. It seems to be working properly.

![Screenshot_20230704_143210](https://github.com/gpbl/react-day-picker/assets/8618014/d5f4958f-b7f8-4ec0-94a6-7644d98bbfa9)

My method is simple, I have installed date-fns-jalali library and give it the alias name date-fns. In this case, the components use the jalali version of date-fns instead of the original one and all dates automatically converted to jalali calendar.

But there is a problem with peerDependencies of DayPicker. When I tried to install DayPicker with npm install command, this does not understand the alias version and return the below error:

`
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve

npm ERR! Found: react-day-picker@8.7.1
npm ERR! node_modules/react-day-picker
npm ERR!   react-day-picker@"^8.7.1" from the root project

npm ERR! Could not resolve dependency:
npm ERR! react-day-picker@"^8.7.1" from the root project

npm ERR! Conflicting peer dependency: date-fns@2.30.0
npm ERR! node_modules/date-fns
npm ERR!   peer date-fns@"^2.28.0" from react-day-picker@8.8.0
npm ERR!   node_modules/react-day-picker
npm ERR!     react-day-picker@"^8.7.1" from the root project

npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
`

Can we make the date-fns dependency as optional? This will resolve the problem and anyone can use different versions of date-fns based on their needs.

**Comment by gpbl** [comment]
Hi @zfarhad this is an interesting case, thanks for sharing. 

> Can we make the date-fns dependency as optional? This will resolve the problem and anyone can use different versions of date-fns based on their needs.

I'm working on this following the discussion in https://github.com/gpbl/react-day-picker/discussions/1087 - so please have patience! Thanks!

**Comment by gpbl** [comment]
We plan to implement this feature in v9. See #1833.

**Comment by gpbl** [comment]
Support for Jalali language has been added to v9: [react-day-picker.js.org/next/using-daypicker/localization#jalali-calendar](https://react-day-picker.js.org/next/using-daypicker/localization#jalali-calendar).

I would appreciate someone testing it - who can help? Thanks!

**Reply by behnam-hs** [reply]
i did test it on v@9 and its not working yet. can u mention my mistake ? 

```
<Calendar
              locale={faIR}
              dir="rtl"
              showWeekNumber={false}
              lang="faIR"
              mode="range"
              defaultMonth={date?.from}
              selected={date}
              onSelect={setDate}
              numberOfMonths={2}
              formatters={{
                formatDay: (day) => day.toLocaleDateString('fa-IR', { day: 'numeric' }),
                formatWeekdayName: (date) => date.toLocaleDateString('fa-IR', { weekday: 'long' }),
                formatMonthCaption: (date) => date.toLocaleDateString('fa-IR', { month: 'long' }),
              }}
/>
```


![Screenshot from 2024-06-24 15-46-59](https://github.com/gpbl/react-day-picker/assets/41870626/98471882-5058-4ee6-ab31-501868c35738)
